lib/cc/analyzer/formatters/json_formatter.rb

```diff
@@ -3,18 +3,36 @@ module Analyzer
     module Formatters
       class JSONFormatter < Formatter
 
+        def initialize
+          @has_begun = false
+        end
+
         def engine_running(engine)
           @active_engine = engine
           yield
           @active_engine = nil
         end
 
+        def started
+          print "[ "
+        end
+
+        def finished
+          print " ]\n"
+        end
+
         def write(data)
           return unless data.present?
 
           document = JSON.parse(data)
           document["engine_name"] = @active_engine.name
-          puts document.to_json
+
+          if @has_begun
```

Comment:
sorry for quibbles! Just wasn't sure the best way :P maybe there's none. 

_the perfect is the enemy of the good_ 

I'll stick with your suggestion. Currently working on tests.